src/views/bridge-confirmation/bridge-confirmation.view.tsx

```diff
@@ -262,10 +263,20 @@ const BridgeConfirmation: FC = () => {
                   setShouldUpdateOnScreenFee(hasOnScreenFeeChanged);
 
                   setMaxPossibleAmountConsideringFee((oldMaxPossibleAmountConsideringFee) => {
+                    const formattedOldMaxPossibleAmountConsideringFee =
+                      oldMaxPossibleAmountConsideringFee &&
+                      (oldMaxPossibleAmountConsideringFee.isNegative()
+                        ? "0"
+                        : formatTokenAmount(oldMaxPossibleAmountConsideringFee, token));
+
+                    const formattedNewMaxPossibleAmountConsideringFee =
+                      newMaxPossibleAmountConsideringFee.isNegative()
+                        ? "0"
+                        : formatTokenAmount(newMaxPossibleAmountConsideringFee, token);
+
                     const hasOnScreenAmountChanged =
-                      oldMaxPossibleAmountConsideringFee !== undefined &&
-                      formatTokenAmount(oldMaxPossibleAmountConsideringFee, token) !==
-                        formatTokenAmount(newMaxPossibleAmountConsideringFee, token);
+                      formattedOldMaxPossibleAmountConsideringFee !==
+                      formattedNewMaxPossibleAmountConsideringFee;
```

Comment:
Why do we need to format both `oldMaxPossibleAmountConsideringFee` and `newMaxPossibleAmountConsideringFee` to check if the amount has changed? We can just use the `.eq` method in `BigNumber` to check if they are equal or not.

Reply:
There are 2 reasons for this:
* When the value is negative we use `"0"` as the formatted value, so two different negative values will format the same.
* The formatted value is using less precision than the actual number so two different positive values can format the same.